src/js/components/ValidationProgress.jsx

```diff
@@ -1,69 +1,61 @@
-      if(code <= STATUS.UPLOADING) indicator = <span className="progress progress-running"></span>
-      if(code >= STATUS.UPLOADED) {
-        indicator = <span className="progress progress-success"></span>
-      }
-      break;
-    case 'parse':
-      if(code === STATUS.PARSING) indicator = <span className="progress progress-running"></span>
-      if(code === STATUS.PARSED_WITH_ERRORS) indicator = <span className="progress progress-error"></span>
-      if(code >= STATUS.PARSED) indicator = <span className="progress progress-success"></span>
-      break;
-    case 'validate':
-      if(code === STATUS.VALIDATING) indicator = <span className="progress progress-running"></span>
-      if(code === STATUS.VALIDATED_WITH_ERRORS) indicator = <span className="progress progress-error"></span>
-      if(code >= STATUS.VALIDATED) indicator = <span className="progress progress-success"></span>
-      break;
-  }
+  if(props.code === STATUS.PARSED_WITH_ERRORS) className += ' error'
 
-  return indicator
-}
+  if(props.code <= STATUS.PARSING) fillWidth = props.percentUploaded/3
+  if(props.code >= STATUS.PARSED) fillWidth = 50
+  if(props.code === STATUS.PARSED_WITH_ERRORS ||
+     props.code > STATUS.VALIDATING) fillWidth = 100
 
-const getUploadStatus = (code) => {
-  if(code <= STATUS.UPLOADING) return <li>{getIndicator(code, 'upload')} Uploading ...</li>
-  return <li>{getIndicator(code, 'upload')} Upload complete</li>
+  return <div className={className} style={{width:fillWidth+'%'}}></div>
 }
 
-const getParsingStatus = (code) => {
-  let textClass = ''
-  if(code < STATUS.PARSING) textClass = 'text-gray-light'
-  if(code <= STATUS.PARSING) return <li className={textClass}>{getIndicator(code, 'parse')} Analyzing file format ...</li>
-  if(code === STATUS.PARSED_WITH_ERRORS) return <li className={textClass}>{getIndicator(code, 'parse')} File format analysis complete with errors</li>
-  return <li>{getIndicator(code, 'parse')} File format analysis complete</li>
-}
+export const getText = props => {
+  let text = 'Uploading...'
+  if(props.code >= STATUS.PARSING) text = 'Analyzing file format...'
+  if(props.code === STATUS.PARSED_WITH_ERRORS) text = 'Invalid file format'
+  if(props.code === STATUS.VALIDATING) text = 'Validating edits...'
+  if(props.code > STATUS.VALIDATING) text = 'Edit validation complete'
 
-const getValidationStatus = (code) => {
-  let textClass = ''
-
-  if(code < STATUS.VALIDATING) textClass = 'text-gray-light'
-  if(code === STATUS.PARSED_WITH_ERRORS) return <li className={textClass}>{getIndicator(code, 'validate')} Edit verification process will not run due to parsing errors</li>
-  if(code <= STATUS.VALIDATING) return <li className={textClass}>{getIndicator(code, 'validate')} Verifying edits ...</li>
-  return <li>{getIndicator(code, 'validate')} Edit verification complete</li>
+  return (
+    <div className="progressText">
+      <span>{text}</span>
+      {props.code === STATUS.VALIDATED_WITH_ERRORS ?
```

Comment:
I don't think you'd have to repeat the span if you did something like:
`if(props.code > STATUS.VALIDATED_WITH_ERRORS) text = <strong>Edits found, review required.</strong>`

But if it's not to signal that edits were found maybe we could just remove that bit. It looks strange to mean to have 2 sentence fragments next to each other and 1 bolded.

Maybe we could just modify the `NavButton` to get that messaging across on the upload page?